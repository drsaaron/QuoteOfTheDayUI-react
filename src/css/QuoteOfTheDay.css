/*
To change this license header, choose License Headers in Project Properties.
To change this template file, choose Tools | Templates
and open the template in the editor.
*/
/* 
    Created on : Dec 28, 2015, 12:31:25 PM
    Author     : scott
*/


root { 
    display: block;
}

body { font-family: Helvetica; font-size: 90% }

#pageBody { position: relative; height: 80%; width: 100%; overflow: auto }

#pageFooter { width: 100% ; height: 10%;  }

#ApplicationCreditSection { font-family: Tahoma; font-size: 85%; color: gainsboro;  }
#ApplicationCreditSection hr { }
#ApplicationCreditSection a:visited { color: gainsboro; }

.QuoteText { font-family: Times New Roman; }
.QuoteSource { font-family: Times New Roman; font-size: 90% }

#pageHeader { float: top; font-size: 125%; color: red; font-family: Tahoma; background-color: blue; }
#pageHeader td { color: red; font-size: 125%; padding: 5px 15px 5px 5px }

.overflow {
    height: 200px;
}

.dateEntry { }

#quoteSourceCodeList { width: 50%; height: 80%; float: left; overflow: auto }
#quotesForSourceCode { width: 50%; height: 80%; float: left; overflow: auto }

.quoteSourceListText {}

.quoteSourceCodeText { font-style: italic }
#quotesForSourceCodeName { font-weight: bold; font-size: 120% }

.addQuoteButton {}
.submit {}
