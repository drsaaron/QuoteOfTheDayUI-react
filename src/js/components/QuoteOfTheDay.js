/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

import {useState} from 'react';
import QuoteText from './QuoteText';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

const QuoteOfTheDay = (props) => {

    const [runDate, setRunDate] = useState(props.quoteOfTheDay.runDate);
    const maxDate = new Date();

    const handleSourceClick = (event) => {
        console.log("getting quotes for " + props.quoteOfTheDay.sourceCode.number);

        event.preventDefault();
        props.retrieveQuotesForSourceCode(props.quoteOfTheDay.sourceCode);
    }
    
    const showDetails = (event) => {
        event.preventDefault();
        props.showDetails(props.quoteOfTheDay.qotd.quoteNumber, props.navigate);
    }
    
    const handleDateChange = (date) => {
	setRunDate(date);
        getQuoteOfTheDay(date);
    }

    const getQuoteOfTheDay = (runDate) => {
        console.log("getting quote of the day for " + runDate);
        props.retrieveQuoteOfTheDay(runDate);
    }

    var quoteNumber = props.quoteOfTheDay.qotd.quoteNumber || -1;
    
    return (
        <div id="quoteOfTheDay">
            <h2>Quote of the day</h2>
            <DatePicker maxDate={maxDate} selected={runDate} onChange={handleDateChange} />
            Quote #<a href="/" onClick={showDetails}>{quoteNumber}</a>
            <QuoteText quote={props.quoteOfTheDay.quote} />
            <div className="QuoteSource">Source: <em><a href="/" onClick={handleSourceClick}>{props.quoteOfTheDay.sourceCode.text}</a></em></div>
        </div>
    );
}

export default QuoteOfTheDay;
