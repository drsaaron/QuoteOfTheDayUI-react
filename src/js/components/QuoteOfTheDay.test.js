import React from 'react';
import renderer from 'react-test-renderer';

import QuoteOfTheDay from './QuoteOfTheDay';

describe('header', () => {
    test('basic', () => {
	const qotd = {
	    runDate: 1000,
	    sourceCode: {
		number: 10,
		text: 'test source'
	    },
	    quote: {
		number: 5,
		text: 'I am a quote'
	    },
	    qotd: {
		quoteNumber: 5
	    }
	};
	const component = renderer.create(<QuoteOfTheDay quoteOfTheDay={qotd} login={{token: "schmo"}} />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
    });
});
