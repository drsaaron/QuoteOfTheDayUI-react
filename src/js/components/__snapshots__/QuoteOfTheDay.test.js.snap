// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header basic 1`] = `
<div
  id="quoteOfTheDay"
>
  <h2>
    Quote of the day
  </h2>
  <div
    className="react-datepicker-wrapper"
  >
    <div
      className="react-datepicker__input-container"
    >
      <input
        className=""
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        readOnly={false}
        type="text"
        value="12/31/1969"
      />
    </div>
  </div>
  Quote #
  <a
    href="/"
    onClick={[Function]}
  >
    5
  </a>
  <div
    className="QuoteText"
    dangerouslySetInnerHTML={
      Object {
        "__html": "I am a quote",
      }
    }
  />
  <div
    className="QuoteSource"
  >
    Source: 
    <em>
      <a
        href="/"
        onClick={[Function]}
      >
        test source
      </a>
    </em>
  </div>
</div>
`;
